src/frontend/src/optimizer/plan_node/stream_dynamic_filter.rs

```diff
@@ -136,12 +132,11 @@ impl_plan_tree_node_for_binary! { StreamDynamicFilter }
 impl StreamNode for StreamDynamicFilter {
     fn to_stream_prost_body(&self, state: &mut BuildFragmentGraphState) -> NodeBody {
         use generic::dynamic_filter::*;
-        let condition = self.core.predicate().as_expr_unless_true().map(|x| {
-            self.base
-                .ctx()
-                .expr_with_session_timezone(x)
-                .to_expr_proto()
-        });
+        let condition = self
```

Comment:
We are not considering the dynamic filter condition an expr as it has to be of a rigid form i.e. not rewritable.